### PR TITLE
Add py2bexp CLI tool to convert qlassf functions to boolean expressions

### PR DESCRIPTION
**Pull request by tomv42**
**Summary:**

This pull request introduces the `py2bexp` command-line tool to the `qlasskit` package. The tool converts qlassf functions in a Python script to boolean expressions. It also includes initial tests for the functionality.

**Changes:**

1. **Added `py2bexp.py` to `qlasskit/tools` package:**
   - Implements the conversion of qlassf functions to boolean expressions.
   - Supports reading from a file or stdin.
   - Allows specifying entrypoint function, output file, expression form, and output format.
   - Includes command-line arguments parsing with `argparse`.

2. **Updated `setup.py`:**
   - Added `py2bexp` as a console script entry point for seamless installation and usage.

3. **Initial Tests in `test_tools.py`:**
   - Included tests for help and version commands.
   - Added tests for default behavior, specific entrypoint, and different expression forms.
   - Verified DIMACS format output for CNF.
   - **Note**: Tests currently do not have assertions but printing results looks good.
       - I don't know what kind of assertion would be appropriate...
   
**Related Issue:**

This PR resolves issue #28

@dakk, I'm looking forward to your feedback.

### CODE REVIEW

**Comment by dakk** [comment]
Thank you, I will review tomorrow UTC; in the meanwhile, can you please take care of failing tests on CI?

**Comment by tomv42** [comment]
Yes, I'm working on it.

**Comment by tomv42** [comment]
I added satisfactory assertions to all the unit tests except one, by comparing the results against sympy expressions and testing that the form is the right one with sympy too.

This didn't work for anf where calling `is_anf` resulted in a type error. Which is weird because it didn't for the others.

For the DIMACS test, as the order of the symbols when converting to DIMACS is not guaranteed, and all the permutations of the names of the symbols give equivalent expressions, the test compares all the possible permutations of the 3 symbols.

File: test/test_tools.py [hunk 1]
```diff
@@ -1,34 +1,249 @@
...
+                "cnf",
+                "-t",
+                "dimacs",
+            ]
+        )
+        print(result.stdout)
+        expr = load(result.stdout)
+        symbols = expr.free_symbols
+        x, y, z = symbols
+        expected1 = And(Or(x, y, Not(z)), Or(z, Not(y)))
```
Comment (dakk) [inline test/test_tools.py:224]: here it is better to use itertools.permutations:

```python
from itertools import permutations

assert_val = False
for sl in permutations(symbols):
   exp = And(Or(sl[0], sl[2], Not(sl[1])), Or(sl[1], Not(sl[2])))
   assert_val = assert_val or expr.equals(exp)

self.assertTrue(assert_val)
```

something like this

File: qlasskit/tools/py2bexp.py [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+
+def read_input(input_file):
+    if input_file == "-":
+        return sys.stdin.read()
+    with open(input_file, "r") as file:
+        return file.read()
+
+
+def find_last_qlassf(qlassf_list):
```
Comment (dakk) [inline qlasskit/tools/py2bexp.py:35]: move this function to tools.py

File: test/test_tools.py [hunk 3]
```diff
@@ -1,34 +1,249 @@
...
+
+    def test_help(self):
+        result = self.run_command(["python", "-m", "qlasskit.tools.py2bexp", "--help"])
+        print(result.stdout)
+
+    def test_version(self):
+        result = self.run_command(
+            ["python", "-m", "qlasskit.tools.py2bexp", "--version"]
+        )
+        print(result.stdout)
```
Comment (dakk) [inline test/test_tools.py:80]: remove this and other prints from test

**Review by dakk — CHANGES_REQUESTED** [review]
Excellent work, I propose you some minor changes and then I can merge it.

**Comment by dakk** [comment]
@tomv42 with very few modification, you can also write the code for this: https://github.com/dakk/qlasskit/issues/26 , if you have time

**Review by dakk — APPROVED** [review]

**Pull request merged**